### PR TITLE
SFO-67: Create navbar

### PR DESCRIPTION
**Pull request by josh231101**
==FEATURES==
Create a simple navbar with two buttons both of them with an event listener

### CODE REVIEW

File: josue/css/style.css [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  margin: 0 16px;
+}
+.btn {
+  border: none;
+  border-radius: 8px;
+  padding: 8px 16px;
+  cursor: pointer;
+}
+.btn--primary {
+  background: var(--blue);
```
Comment (frank-martinez-iest) [inline josue/css/style.css:58]: 👌

File: josue/js/app.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const signInBtn = document.querySelector("#sign");
+const register = document.querySelector("#register");
+
+signInBtn.addEventListener("click",e=>{
+    console.log(e)
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: You don't need to print `e` at the moment

File: josue/js/app.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    console.log("You clicked sign in !!!!!")
+})
+register.addEventListener("click",e=>{
+    console.log(e);
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: You don't need to print `e` at the moment

**Review by frank-martinez-iest — CHANGES_REQUESTED** [review]
Looks good to me! Left a couple comments though

File: josue/js/app.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+const submitBtn = document.querySelector("#submit");
+const registerFields = document.querySelector("#register");
+let fields = ['name','phone'];
+
+registerFields.remove();
+signInBtn.addEventListener("click",e=>{
+    registerFields.classList.add("register--hidden");
+    registerFields.remove();
+    submitBtn.innerText = "Sign In"
+    fields = ['name','phone']
```
Comment (frank-martinez-iest) [inline josue/js/app.js:15]: you already assigned this array to the variable `fields` on line 8, why don't you make it a constant and delete this line? On line 16 you will still have a reference to the right array

File: josue/js/app.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    validator.updateFields(fields,"signin")
+
+})
+registerBtn.addEventListener("click",e=>{
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: Either you are missing `e.preventDefault()` or if no needed then you can remove the `e` as an argument

File: josue/js/app.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    form.insertBefore(registerFields,submitBtn);
+    registerFields.classList.remove("register--hidden");
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: can you add a better name to the class register? in here I don't know if you are hiding an input, a form or a button

File: josue/js/app.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    submitBtn.innerText = "Register"
+    const newFields= ['email','password','password_confirmation']
+    validator.updateFields(newFields,"register")
+});
+submitBtn.addEventListener('click',e=>{
+    console.log("click");
+    e.preventDefault();
+    if(validator.validateOnSubmit()){
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: You must store the return value of `validator.validateOnSubmit()` into a variable that tells me what does the boolean return value mean, for example `const isFormValid = validator.validateOnSubmit()` and now you can do `if(isFormValid){...` which is WAY more readable, right?

File: josue/js/app.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+        // Fields are filled correctly now we check if user wants to register or login
+        if(submitBtn.innerText == 'Sign In'){
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: NIT: Is good practice to store all strings in a variable at the top so if something changes is easier to locate the string at the top of the js file instead of going through all the places we are making reference to it, so you can do `const SIGN_IN = 'Sign In'`

File: josue/js/app.js [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+            const loginUser = validator.createObjectFromForm()
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: `const userData = validator.parseFormData()` is a more accurate name.

File: josue/js/app.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+                
+            }else{
+                alert("You are not Registerd, login instead?")
+            }
+        }
+        else{
+            // User wants to register.. save it to localstorage
+            const  registeredUser = validator.createObjectFromForm();
+            console.log(registeredUser);
+            window.localStorage.setItem(`${registeredUser.phone}`,JSON.stringify(registeredUser));
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: 👌

**Review by frank-martinez-iest — CHANGES_REQUESTED** [review]
Nice approach, but I don't like that there are too many changes unrelated to the ticket scope. This is a sign of a junior developer, not limiting their PRs to the scope agreed on the tickets. Please make sure in the future your PRs ONLY cover the scope of the ticket

File: josue/js/app.js (hunk 4 above)
Comment (josh231101) [reply josue/js/app.js:15]: Right, I'll change it to a const. The reason why is reasigned is because the user can change from the register form and then getting back to login, if we don't refresh the fields the event listeners of the register form will be active.

File: josue/js/app.js (hunk 7 above)
Comment (josh231101) [reply josue/js/app.js]: Yes I'll change it :100:

File: josue/js/Formvalidator.js [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+            field.classList.add("input--incorrect");
+            errormessage.innerText = message;
+        }
+    }
+    updateFields(fields,formType){
+        if(formType === 'register'){
+            // If the current fields are less than combining the current fields with the new fields and deleting repeated elements
+            // That means that we haven't yet assigned the event listeners to the register form inputs
+
+            if(this.fields.length < [...new Set([...this.fields,...fields])].length){
```
Comment (frank-martinez-iest) [inline josue/js/Formvalidator.js]: if(this.fields.length < [...this.fields, ...fields].length

File: josue/js/app.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+const signInBtn = document.querySelector("#signBtn");
+const registerBtn = document.querySelector("#registerBtn");
+const submitBtn = document.querySelector("#submit");
+const registerFields = document.querySelector("#register-fields");
+const boardWrapper = document.querySelector(".account__board-wrapper");
+const userDashboard = document.querySelector(".account__board-user");
+const welcomeSplash = document.querySelector('.account__splash-img');
+const accountSection =document.querySelector(".account");
+const fields = ['name','phone'];
+
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: const FORM_TYPES = { SIGN_IN: "signig", REGISTER: "register"}

File: josue/css/style.css [hunk 13]
```diff
@@ -142,6 +142,15 @@ input {
...
+    grid-template-rows: 1fr 1fr;
+  }
+  .navbar {
+    padding: 20px;
+  }
+  .account__left-wrapper {
+    padding: 20px;
+  }
+  .account__board-wrapper {
+    padding: 20px;
```
Comment (frank-martinez-iest) [inline josue/css/style.css]: Values must consistently be numbers divisible by 8

File: josue/css/style.css [hunk 14]
```diff
@@ -142,6 +142,15 @@ input {
...
   .account {
     grid-template-columns: 100%;
-    grid-template-rows: 220px 50%;
+    grid-template-rows: 1fr 1fr;
+  }
+  .navbar {
+    padding: 20px;
+  }
+  .account__left-wrapper {
+    padding: 20px;
```
Comment (frank-martinez-iest) [inline josue/css/style.css]: Values must consistently be numbers divisible by 8

File: josue/css/style.css [hunk 15]
```diff
@@ -142,6 +142,15 @@ input {
 @media screen and (max-width: 768px) {
   .account {
     grid-template-columns: 100%;
-    grid-template-rows: 220px 50%;
+    grid-template-rows: 1fr 1fr;
+  }
+  .navbar {
+    padding: 20px;
```
Comment (frank-martinez-iest) [inline josue/css/style.css]: Values must consistently be numbers divisible by 8

File: josue/js/Formvalidator.js [hunk 16]
```diff
@@ -73,36 +73,7 @@ export default class FormValidator{
...
-                // We have to delete the event listeners given to the inputs in register
-                const registerFields = [...fields]
-                registerFields.splice(2).forEach(field=>{
-                    const input = document.querySelector(`#${field}`)
-                    input.removeEventListener('input');
-                })
-            }
-        }
-    }
     parseFormData(){
```
Comment (frank-martinez-iest) [inline josue/js/Formvalidator.js]: We agreed this method is not related to form validation and needed to be moved outside this function right?

File: josue/js/app.js [hunk 17]
```diff
@@ -1,37 +1,49 @@
...
 const SIGN_IN = 'Sign In';
-registerFields.remove();
-boardWrapper.remove();
+const REGISTER = 'Register';
+
+const addRegisterFieldsToValidator = () =>{
+    // First step, combine the actual validator fields with the registerFields
+    const fieldsCombined = [...validator.fields,...registerFieldsIds];
+    // We have to filter the array to delete the repeated elements 
+    const updatedFormFields = fieldsCombined.filter((field,i)=> fieldsCombined.indexOf(field) === i);
```
Comment (frank-martinez-iest) [inline josue/js/app.js]: if the only thing we are doing here is removing duplicates then you can use `new Set` as you did before only following the advice seen here in the 2nd answer https://stackoverflow.com/questions/9229645/remove-duplicate-values-from-js-array/9229821

**Review by frank-martinez-iest — CHANGES_REQUESTED** [review]
Good job, there were a couple of things that still need to be updated. Let me know if you have any question please.

File: josue/js/Formvalidator.js (hunk 16 above)
Comment (josh231101) [reply josue/js/Formvalidator.js]: Sorry, I thought that it was part of it.. I'll move it jeje

File: josue/js/app.js [hunk 18]
```diff
@@ -19,15 +19,23 @@ const addRegisterFieldsToValidator = () =>{
     // First step, combine the actual validator fields with the registerFields
     const fieldsCombined = [...validator.fields,...registerFieldsIds];
     // We have to filter the array to delete the repeated elements 
-    const updatedFormFields = fieldsCombined.filter((field,i)=> fieldsCombined.indexOf(field) === i);
-    validator.fields = updatedFormFields;
+    const newFilteredFormArray = [...new Set(fieldsCombined)]
```
Comment (frank-martinez-iest) [inline josue/js/app.js:22]: 👌

**Review by frank-martinez-iest — APPROVED** [review]
APPROVED
